src/main/java/net/sf/jabref/logic/autocompleter/NameFieldAutoCompleter.java

```diff
@@ -86,14 +79,16 @@ public boolean isSingleUnitField() {
     }
 
     @Override
-    public void addBibtexEntry(BibEntry entry) {
+    public void addToIndex(BibEntry entry) {
         if (entry == null) {
             return;
         }
+
         for (String fieldName : fieldNames) {
-            if (entry.hasField(fieldName)) {
-                String fieldValue = entry.getField(fieldName);
-                AuthorList authorList = AuthorList.parse(fieldValue);
+            Optional<String> fieldValue = entry.getFieldOptional(fieldName);
+
+            if (fieldValue.isPresent()) {
```

Comment:
Hm I don't like the code readability with ifPresent() because the body includes several line sof code.